core/app/models/spree/variant/pricing_options.rb

```diff
@@ -33,7 +33,7 @@ def self.default_price_attributes
       def self.from_line_item(line_item)
         tax_address = line_item.order.try!(:tax_address)
         new(
-          currency: line_item.order.try(:currency) || line_item.currency || Spree::Config.currency,
+          currency: line_item.order.try(:currency) || Spree::Config.currency,
```

Comment:
With the delegate, this can actually be written shorter as `line_item.currency` :)